cpp/dolfin/common/MPI.cpp

```diff
@@ -17,7 +20,8 @@ dolfin::MPI::Comm::Comm(MPI_Comm comm)
   {
     int err = MPI_Comm_dup(comm, &_comm);
     if (err != MPI_SUCCESS)
-      log::error("Duplication of MPI communicator failed (MPI_Comm_dup");
+      spdlog::error("Duplication of MPI communicator failed (MPI_Comm_dup)");
+      throw std::runtime_error("Duplication of MPI communicator failed (MPI_Comm_dup)");
```

Comment:
The old design logged and threw. The new design requires separate steps. Thoughts?

Reply:
I think it looks ok - bare in mind that the logger will (or should) be used for much more than just error logging.

Do we have any indication of what is 'good' practice?